source/common/config/config_provider_impl.h

```diff
@@ -219,16 +219,7 @@ class ConfigSubscriptionCommonBase : protected Logger::Loggable<Logger::Id::conf
    * @param complete_cb the callback to run when the update propagation is done.
    */
   void applyConfigUpdate(
-      const ConfigUpdateCb& update_fn, const Event::PostCb& complete_cb = []() {}) {
-    tls_->runOnAllThreads(
-        [this, update_fn]() {
-          // NOTE: there is a known race condition between *this* subscription being teared down in
-          // main thread and the posted callback being executed before the destruction. See more
-          // details in https://github.com/envoyproxy/envoy/issues/7902
-          tls_->getTyped<ThreadLocalConfig>().config_ = update_fn(getConfig());
-        },
-        complete_cb);
-  }
+      const ConfigUpdateCb& update_fn, const Event::PostCb& complete_cb = []() {});
```

Comment:
Is the second param actually used anywhere here? I don't see it? Can we remove it?

Reply:
let me remove it.
It's a over-design now that we know it can't be used to guarantee TLS owner outlives update callbacks.